VMAccess/README.md

```diff
@@ -0,0 +1,252 @@
+one in section 2.1 and 2.2 below.
+ 
+### 2.1. Using [**Azure CLI**][azure-cli]
+Before deploying VMAccess Extension, you should configure your `protected.json`
+(in section 1.2 above).
+
+#### 2.1.1 Classic
+The Classic mode is also called Azure Service Management mode. You can change to it by running:
+```
+$ azure config mode asm
+```
+
+You can deploying VMAccess Extension by running:
+```
+$ azure vm extension set <vm-name> VMAccessForLinux Microsoft.OSTCExtensions <version> \
+--private-config-path protected.json
+```
+
+In the command above, you can change version with `"*"` to use latest
+version available, or `"1.*"` to get newest version that does not introduce non-
+breaking schema changes. To learn the latest version available, run:
+```
+$ azure vm extension list
+```
+
+#### 2.1.2 Resource Manager
+You can change to Azure Resource Manager mode by running:
+```
+$ azure config mode arm
+```
+
+You can deploying VMAccess Extension by running:
+```
+$ azure vm extension set <resource-group> <vm-name> VMAccessForLinux Microsoft.OSTCExtensions <version> \
+--private-config-path protected.json
+```
+
+> **NOTE:** In ARM mode, `azure vm extension list` is not available for now.
+
+
+### 2.2. Using [**Azure Powershell**][azure-powershell]
+
+#### 2.2.1 Classic
+You can change to Azure Service Management mode by running:
+```powershell
+Switch-AzureMode -Name AzureServiceManagement
+```
+
+You can deploying VMAccess Extension by running:
+```powershell
+$VmName = '<vm-name>'
+$vm = Get-AzureVM -ServiceName $VmName -Name $VmName
+
+$ExtensionName = 'VMAccessForLinux'
+$Publisher = 'Microsoft.OSTCExtensions'
+$Version = <version>
+
+$PublicConf = '{}'
+$PrivateConf = '{
+  "username": "<username>",
+  "password": "<password>",
+  "ssh_key": "<pem-cert-contents>",
+  "reset_ssh": true|false,
+  "remove_user": "<username-to-remove>"
+}'
+
+Set-AzureVMExtension -ExtensionName $ExtensionName -VM  $vm -Publisher $Publisher -Version $Version -PrivateConfiguration $PrivateConf -PublicConfiguration $PublicConf | Update-AzureVM
+```
+
+#### 2.2.2 Resource Manager
+You can change to Azure Resource Manager mode by running:
+```powershell
+Switch-AzureMode -Name AzureResourceManager
+```
+
+You can deploying VMAccess Extension by running:
+```powershell
+$RGName = '<resource-group-name>'
+$VmName = '<vm-name>'
+$Location = '<location>'
+
+$ExtensionName = 'VMAccessForLinux'
+$Publisher = 'Microsoft.OSTCExtensions'
+$Version = <version>
+
+$PublicConf = '{}'
+$PrivateConf = '{
+  "username": "<username>",
+  "password": "<password>",
+  "ssh_key": "<pem-cert-contents>",
+  "reset_ssh": true|false,
+  "remove_user": "<username-to-remove>"
+}'
+
+Set-AzureVMExtension -ResourceGroupName $RGName -VMName $VmName -Name $ExtensionName -Publisher $Publisher -ExtensionType $ExtensionName -TypeHandlerVersion $Version -Settingstring $PublicConf -$PublicConf $PrivateConf -Location $Location
```

Comment:
Just a suggestion, would you please break down the long line command into multiple lines?